Tidy error on Bitcoin Core CI

```
[22:21:15.870] /ci_container_base/src/ipc/process.cpp:75:17: error: use starts_with instead of compare() == 0 [modernize-use-starts-ends-with,-warnings-as-errors]
[22:21:15.870]    75 |     if (address.compare(0, 4, "unix") == 0 && (address.size() == 4 || address[4] == ':')) {
[22:21:15.870]       |                 ^~~~~~~~~~~~~       ~~~~~~
[22:21:15.870]       |                 starts_with(        )
[22:21:15.870] 850 warnings generated.
[22:21:15.870] 
[22:21:15.870] ^^^ ⚠️ Failure generated from clang-tidy
```

https://github.com/bitcoin/bitcoin/pull/30975/checks?check_run_id=34105951022

--- a/src/ipc/process.cpp
+++ b/src/ipc/process.cpp
@@ -72,7 +72,7 @@ static bool ParseAddress(std::string& address,
                   struct sockaddr_un& addr,
                   std::string& error)
 {
-    if (address.compare(0, 4, "unix") == 0 && (address.size() == 4 || address[4] == ':')) {
+    if (address == "unix" || address.starts_with("unix:")) {
         fs::path path;
         if (address.size() <= 5) {
             path = data_dir / fs::PathFromString(strprintf("%s.sock", RemovePrefixView(dest_exe_name, "bitcoin-")));

